iree/turbine/kernel/wave/analysis/index_sequence_analysis.py

```diff
@@ -476,15 +479,18 @@ def propagate_indices(
                 source, source_index, source_vector_shapes, symbolic_constraints
             ):
                 continue
-            source_index = source.transform_index(source_index)
-            source.index = combine_indices(source.index, source_index)
+            # GetResults inherit their index from the Iterate node
+            # and hence we don't need to update their index.
+            if not isinstance(source, GetResult):
+                source_index = source.transform_index(source_index)
+                source.index = combine_indices(source.index, source_index)
             source.vector_shapes = source_vector_shapes
             append_aliased_shapes(source, symbolic_constraints)
         visited.add(source)
         for func in [get_inputs, get_users]:
-            sources, reduction = add_nodes_to_sources(
```

Comment:
seems like we don't need reductio in add_nodes_to_sources at all?

Reply:
yes I can remove it.